lib/lumberjack/beats/server.rb

```diff
@@ -66,8 +66,16 @@ def run(&block)
         # unless the server is closing
         next unless connection
 
+        Thread.abort_on_exception = true
         Thread.new(connection) do |connection|
-          connection.run(&block)
+          begin
+            connection.run(&block)
+          rescue Lumberjack::Beats::Connection::ConnectionClosed 
+            # Connection will raise a wrapped exception upstream, 
+            # but if the threads are managed by the library we can simply ignore it.
```

Comment:
On the plugin side we need to know when something close a connection, before that we were simply ignoring it and closing the socket. no feedback was ever given back to the library which mean we could have lost events inside a codec. :(